robot/board.py

```diff
@@ -155,8 +155,11 @@ def close(self):
         """
         self.socket.detach()
 
+    def __repr__(self):
+        return "<{}>".format(self)
```

Comment:
I suggest that this be:
``` python
    return '{}({!r}'.format(self.__class__.__name__, self.serial)
```
That would be truer `repr` behaviour, though would admittedly rely on classes which take other construction parameters overriding the default.

If we do that, we can then drop the `__str__` entirely.

Reply:
I disagree, this would not be truer `repr` behaviour. Please see my above comment and link to the Python 3 documentation.

Reply:
Damn. Apparently I meant `self.socket_path`, not `self.serial`.

Reply:
In which case we might want to keep the `__str__`.